xl2times/transforms.py

```diff
@@ -1246,17 +1244,25 @@ def complete_commodity_groups(
     model: datatypes.TimesModel,
 ) -> Dict[str, DataFrame]:
     """
-    Complete the list of commodity groups
+    Complete the list of commodity groups.
     """
 
-    commodities = generate_topology_dictionary(tables, model)[
-        "commodities_by_name"
-    ].rename(columns={"commodity": "commoditygroup"})
-    cgs_in_top = model.topology["commoditygroup"].to_frame()
-    commodity_groups = pd.concat([commodities, cgs_in_top])
-    model.commodity_groups = commodity_groups.drop_duplicates(
-        keep="first"
-    ).reset_index()
+    # Single member CGs i.e., CG and commodity are the same
+    single_cgs = (
+        model.commodities[["region", "commodity"]]
+        .drop_duplicates(ignore_index=True)
+        .copy()
+    )
+    single_cgs["commoditygroup"] = single_cgs["commodity"]
+    # Commodity groups from topology
+    top_cgs = (
+        model.topology[["region", "commodity", "commoditygroup"]]
+        .drop_duplicates(ignore_index=True)
+        .copy()
```

Comment:
```suggestion
        .drop_duplicates(ignore_index=True)
```
You don't need to copy here either.